contracts/NewElection.sol

```diff
@@ -61,6 +61,15 @@ contract NewElection {
       return currentElection.candidatesList[candidateID].voteCount;
     }
 
+    function displayOwnVote() constant returns (string candidateName) {
+      Voter currentVoter = voters[msg.sender];
+
+      //if(currentVoter.voted != true) throw;
+			uint candidateID = currentVoter.votedFor;
+			bytes32 candidate32 = currentElection.candidatesList[candidateID].name;
+			return bytes32ToString(candidate32);
```

Comment:
So as we use the bytes32ToString does this mean we should keep this function in the contract? Or did we want to move it further up the stack to the index.js file in a later refactor?